File: leet/src/Q_11_20/Q11/Q11.go
package main

import "math"

//执行用时 :16 ms, 在所有 Go 提交中击败了95.81%的用户
//内存消耗 :5.7 MB, 在所有 Go 提交中击败了64.91%的用户
func maxArea(height []int) int {
	start := 0
	end := len(height) - 1
	tem := float64(0)
	max := float64(0)
	for start != end {
		tem = (math.Min(float64(height[start]), float64(height[end]))) * float64(end-start)
		max = math.Max(tem, max)
		if height[start] <= height[end] {
			start++
		} else {
			end--
		}
	}
	return int(max)
}

File: leet/src/Q_11_20/Q11/Q11.py
class Solution(object):
    def maxArea(self, height):
        # 140 ms 在所有 Python 提交中击败了 51.74% 的用户
        # 12.9 MB 在所有 Python 提交中击败了 43.26% 的用户
        start = 0
        end = len(height) - 1
        tmax = 0
        while start != end:
            tem = int(min(height[start], height[end]) * (end - start))
            tmax = max(tem, tmax)
            if height[start] <= height[end]:
                start = start + 1
            else:
                end = end - 1
        return tmax


if __name__ == '__main__':
    print(Solution().maxArea([2, 3, 10, 5, 7, 8, 9]))
